File: growing-shop-java/src/main/java/com/example/growingshop/domain/auth/api/PolicyController.java
package com.example.growingshop.domain.auth.api;

import com.example.growingshop.domain.auth.dto.RoleRequest;
import com.example.growingshop.domain.auth.dto.RoleResponse;
import com.example.growingshop.domain.auth.service.PolicyService;
import lombok.RequiredArgsConstructor;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

@RestController
@RequestMapping("/policies")
@RequiredArgsConstructor
public class PolicyController {
    private final PolicyService policyService;

    @PostMapping
    public ResponseEntity<RoleResponse.PoliciesRes> create(@RequestBody RoleRequest.CreatePolicy req) {
        return new ResponseEntity<>(
                RoleResponse.PoliciesRes.from(policyService.create(req)),
                HttpStatus.OK
        );
    }

    @DeleteMapping
    public void delete(Long id) {
        policyService.delete(id);
    }
}

File: growing-shop-api/auth/src/main/kotlin/com/example/growingshopauth/auth/api/PolicyController.kt
package com.example.growingshopauth.auth.api

import com.example.growingshopauth.auth.dto.RoleRequest
import com.example.growingshopauth.auth.dto.RoleResponse
import com.example.growingshopauth.auth.service.PolicyService
import org.springframework.http.HttpStatus
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.*

@RestController
@RequestMapping("/policies")
class PolicyController(
    private val policyService: PolicyService
) {

    @PostMapping
    fun create(@RequestBody req: RoleRequest.CreatePolicy): ResponseEntity<RoleResponse.PoliciesRes> = ResponseEntity(
        RoleResponse.PoliciesRes.from(policyService.create(req)),
        HttpStatus.OK
    )

    @DeleteMapping("/{id}")
    fun delete(@PathVariable id: Long): ResponseEntity<Any> {
        policyService.delete(id)

        return ResponseEntity(HttpStatus.NO_CONTENT)
    }
}
